### PR TITLE
Add core files.

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

**Comment by mattcollier** [comment]
@dlongley a couple fixmes in there having to do with achieving parity between the two implementations.

**Comment by dmitrizagidulin** [comment]
+1 to this PR in particular, but I'm uneasy about the existence of this repo. Not a blocker, tho, I need to run a couple of Webpack size experiments.

File: lib/base64url.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    line += _alphabet.charAt(chr1 >> 2);
+    line += _alphabet.charAt(((chr1 & 3) << 4) | (chr2 >> 4));
+    if(!isNaN(chr2)) {
+      line += _alphabet.charAt(((chr2 & 15) << 2) | (chr3 >> 6));
+      if(!isNaN(chr3)) {
+        line += _alphabet.charAt(chr3 & 63);
+      }
+    }
+  }
+  output += line;
```
Comment (dlongley) [inline lib/base64url.js]: We can remove this line and replace the `line` var everywhere with `output` and then remove the `line` var it looks like.

File: lib/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const env = require('./env');
+
+const api = {};
+module.exports = api;
+
+let base64url;
+if(env.nodejs) {
+  // FIXME: the `encode` API for npm base64url allows string or buffer, should
+  // steps be taken homogenize the APIs in different libs, or consider it a
+  // feature?
```
Comment (dlongley) [inline lib/index.js]: We should at least throw if `input` is not a Uint8Array/Buffer in the js-only version if we don't want to do something to support string inputs.

**Review by dlongley — CHANGES_REQUESTED** [review]

File: lib/index.js (hunk 2 above)
Comment (mattcollier) [reply lib/index.js]: done.

**Comment by mattcollier** [comment]
@dlongley @davidlehn OK, both `encode` implementations accept bytes/strings and both `decode` implementations return bytes.

**Review by dlongley — APPROVED** [review]

**Review by davidlehn — APPROVED** [review]
Couple things should be done now, or issues added: update README with docs on the API, update the main doc strings so the input/output formats are clearer, add keywords to package.json.

**Comment by mattcollier** [comment]
@davidlehn what keyword would you like to see?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  },
+  "scripts": {
+    "test": "npm run test-node",
+    "test-node": "cross-env NODE_ENV=test mocha --preserve-symlinks -t 30000 -A -R ${REPORTER:-spec} tests/*.spec.js",
+    "test-karma": "karma start",
+    "benchmark": "node benchmark/benchmark.js",
+    "coverage": "cross-env NODE_ENV=test nyc --reporter=lcov --reporter=text-summary npm run test-node",
+    "coverage-report": "nyc report",
+    "lint": "eslint lib tests"
+  }
```
Comment (davidlehn) [inline package.json]:
```suggestion
  },
  "keywords" : {
    "base64",
    "base64url"
  }
```

File: package.json [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  "scripts": {
+    "test": "npm run test-node",
+    "test-node": "cross-env NODE_ENV=test mocha --preserve-symlinks -t 30000 -A -R ${REPORTER:-spec} tests/*.spec.js",
+    "test-karma": "karma start",
+    "benchmark": "node benchmark/benchmark.js",
+    "coverage": "cross-env NODE_ENV=test nyc --reporter=lcov --reporter=text-summary npm run test-node",
+    "coverage-report": "nyc report",
+    "lint": "eslint lib tests"
+  },
+  "keywords" : {
```
Comment (davidlehn) [inline package.json]: Ooops, typo on that last patch.  This should be an array.

**Pull request merged**